lib/ansible/modules/monitoring/grafana_dashboard.py

```diff
@@ -298,11 +314,20 @@ def grafana_dashboard_changed(payload, dashboard):
 def grafana_create_dashboard(module, data):
 
     # define data payload for grafana API
-    try:
-        with open(data['path'], 'r') as json_file:
-            payload = json.load(json_file)
-    except Exception as e:
-        raise GrafanaAPIException("Can't load json file %s" % to_native(e))
+    payload = {}
+    if re.match(r"^\d+$", data['path']):
+        data['path'] = "https://grafana.com/api/dashboards/%s/revisions/1/download" % data['path']
+    if data['path'].startswith('http'):
+        r, info = fetch_url(module, data['path'])
+        if info['status'] != 200:
+            raise GrafanaAPIException('Unable to download grafana dashboard from url %s : %s' % (data['path'], info))
+        payload = json.loads(r.read())
+    else:
+        try:
+            with open(data['path'], 'r') as json_file:
+                payload = json.load(json_file)
+        except Exception as e:
+            raise GrafanaAPIException("Can't load json file %s" % to_native(e))
```

Comment:
I think you should use new parameters instead of creating aliases on `path`.
It will allow you to properly document the usage of each parameters.

In the current implementation, you cannot provide the revision of the dashboard if you provide an id as the revision is hardcoded to "1" in the url. So you need an additional parameter that specifies the revision.

I guess that `path` could be a local path on disk or an url.
But it should not accept an id for a location on grafana.com. This case should have dedicated parameters for the dashboard id and revision.

Reply:
OK, fixed :)